### DISCUSSION TITLE
サーボモードの0位置の設定方法

### DISCUSSION THREAD

**Discussion by arkwknsk** [ponoor/step-series-support#32] [Q&A] [answered]
サーボモードの"0"の位置が初期化されるのは、通電して起動したタイミングと思っていますが、
その他、OSCコマンド等で現在のモータの位置を0に設定する方法はあるでしょうか？

**Comment by kanta** [comment, marked as answer]
```
/resetPos (int)motorID
```
こちらのメッセージで、指定モータの座標を `0` にできます。
あるいは、以下のメッセージで、 `newPosition` に `0` を指定することによって、現在の座標を `0` に設定できます。
```
/setPosition (int)motorID (int)newPosition
```
`/resetPos` は常時送ることができますが、 `/setPosition` はモータが停止している時のみ送ることができます。
参考： https://ponoor.com/docs/step-series/osc-command-reference/absolute-position-management/
これら2個のメッセージはサーボモードか否かを問わず使用できますが、サーボモードの場合は以下の点に注意が必要です。

- サーボモードで動作しているときは、パラメータ設定によってはごくわずかな動きを繰り返したり、非常に遅いスピードでじりじりと動く挙動になることがあり、モータが必ず停止している状態と言える瞬間がありません。
-  現在の座標をリセットすると、相対的に目標位置が急にずれることになり、その目標位置に向かって突然モータが動き始めてしまいます。

ですので、例えば原点付近で座標をリセットする場合など、目標位置があまりずれないときは単純に `/resetPos` を送れば問題ないこともありますが、一般的にサーボモードで動作中に現在位置をリセットする場合は、以下の順序で行うのが安全です。

1. `/softStop` `/hardStop` などでモータを停止させる。これらのモータ停止メッセージを送るとサーボモードが自動で解除されます。
2. 上記コマンドのどちらかを用いて座標をリセットする。
3. 再度サーボモードをオンにする。

**Reply by arkwknsk** [reply]
@kanta 細かい情報、ありがとうございます
